File: backend/app/schemas/document.py
from pydantic import BaseModel


class DocumentUploadResponse(BaseModel):
    document_id: str
    filename: str
    original_filename: str
    file_type: str
    content_type: str
    size: int
    is_duplicate: bool


class DocumentTextExtractResponse(BaseModel):
    document_id: str
    text_filename: str
    page_count: int
    character_count: int
    preview: str


class DocumentChunkResponse(BaseModel):
    document_id: str
    chunks_filename: str
    chunk_count: int
    chunk_size: int
    chunk_overlap: int
    first_chunk_preview: str


class DocumentIndexResponse(BaseModel):
    document_id: str
    collection_name: str
    indexed_chunk_count: int
    persist_directory: str


class DocumentImportResponse(BaseModel):
    document_id: str
    filename: str
    original_filename: str
    file_type: str
    content_type: str
    size: int
    is_duplicate: bool
    page_count: int
    character_count: int
    text_filename: str
    chunks_filename: str
    chunk_count: int
    indexed_chunk_count: int
    collection_name: str
    knowledge_base_ids: list[str] = []
    ocr_status: str | None = None
    ocr_message: str | None = None


class LocalDocumentImportRequest(BaseModel):
    file_path: str
    knowledge_base_ids: list[str] = []
    chunk_size: int = 1000
    chunk_overlap: int = 200


class DocumentListItem(BaseModel):
    document_id: str
    original_filename: str
    stored_filename: str
    file_type: str
    content_type: str
    size: int
    uploaded_at: str
    updated_at: str
    page_count: int
    character_count: int = 0
    chunk_count: int
    indexed_chunk_count: int = 0
    is_indexed: bool
    file_path: str
    text_path: str | None = None
    chunks_path: str | None = None
    collection_name: str | None = None
    knowledge_base_ids: list[str] = []
    ocr_status: str | None = None
    ocr_message: str | None = None


class DocumentListResponse(BaseModel):
    documents: list[DocumentListItem]
    total: int


class DocumentDetailResponse(DocumentListItem):
    pass


class DocumentDeleteResponse(BaseModel):
    document_id: str
    deleted_metadata: bool
    deleted_files: list[str]
    missing_files: list[str]
    deleted_vector_count: int
    removed_knowledge_base_id: str | None = None
    is_orphan: bool = False


class DocumentFileActionResponse(BaseModel):
    document_id: str
    file_path: str
    action: str
    success: bool


class DocumentPreviewResponse(BaseModel):
    document_id: str
    filename: str
    file_type: str
    preview_supported: bool
    preview_mode: str
    content: str | None = None
    file_url: str | None = None
    page: int | None = None
    chunk_index: int | None = None
    message: str | None = None


class DocumentKnowledgeBaseResponse(BaseModel):
    document_id: str
    knowledge_base_ids: list[str]

File: frontend/src/types/document.ts
export interface DocumentItem {
  document_id: string;
  original_filename: string;
  stored_filename: string;
  file_type: string;
  content_type: string;
  size: number;
  uploaded_at: string;
  updated_at: string;
  page_count: number;
  character_count: number;
  chunk_count: number;
  indexed_chunk_count: number;
  is_indexed: boolean;
  file_path: string;
  text_path?: string | null;
  chunks_path?: string | null;
  collection_name?: string | null;
  knowledge_base_ids: string[];
  ocr_status?: string | null;
  ocr_message?: string | null;
}

export interface DocumentListResponse {
  documents: DocumentItem[];
  total: number;
}

export interface DocumentImportResponse {
  document_id: string;
  filename: string;
  original_filename: string;
  file_type: string;
  content_type: string;
  size: number;
  is_duplicate: boolean;
  page_count: number;
  character_count: number;
  text_filename: string;
  chunks_filename: string;
  chunk_count: number;
  indexed_chunk_count: number;
  collection_name: string;
  knowledge_base_ids: string[];
  ocr_status?: string | null;
  ocr_message?: string | null;
}

export interface DocumentDeleteResponse {
  document_id: string;
  deleted_metadata: boolean;
  deleted_files: string[];
  missing_files: string[];
  deleted_vector_count: number;
  removed_knowledge_base_id?: string | null;
  is_orphan?: boolean;
}

export interface DocumentFileActionResponse {
  document_id: string;
  file_path: string;
  action: string;
  success: boolean;
}

export interface DocumentPreviewResponse {
  document_id: string;
  filename: string;
  file_type: string;
  preview_supported: boolean;
  preview_mode: "pdf" | "text" | "image" | "unsupported";
  content?: string | null;
  file_url?: string | null;
  page?: number | null;
  chunk_index?: number | null;
  message?: string | null;
}

export interface DocumentKnowledgeBaseResponse {
  document_id: string;
  knowledge_base_ids: string[];
}

export interface PendingImportFile {
  id: string;
  name: string;
  size: number;
  lastModified: number;
  file?: File;
  filePath?: string;
}

export type DocumentTypeFilter = "all" | "pdf" | "docx" | "pptx" | "txt" | "md" | "image";
